nautobot/extras/models/jobs.py

```diff
@@ -799,87 +815,150 @@ def enqueue_job(
         if task_queue is None:
             task_queue = job_model.default_job_queue.name
 
-        job_celery_kwargs = {
-            "nautobot_job_job_model_id": job_model.id,
-            "nautobot_job_profile": profile,
-            "nautobot_job_user_id": user.id,
-            "queue": task_queue,
-        }
-
-        if schedule is not None:
-            job_celery_kwargs["nautobot_job_schedule_id"] = schedule.id
-        if job_model.soft_time_limit > 0:
-            job_celery_kwargs["soft_time_limit"] = job_model.soft_time_limit
-        if job_model.time_limit > 0:
-            job_celery_kwargs["time_limit"] = job_model.time_limit
-
-        if celery_kwargs is not None:
-            job_celery_kwargs.update(celery_kwargs)
-
-        if synchronous:
-            # synchronous tasks are run before the JobResult is saved, so any fields required by
-            # the job must be added before calling `apply()`
-            job_result.celery_kwargs = job_celery_kwargs
-            job_result.save()
-
-            # setup synchronous task logging
-            setup_nautobot_job_logging(None, None, app.conf)
-
-            # redirect stdout/stderr to logger and run task
-            redirect_logger = get_logger("celery.redirected")
-            proxy = LoggingProxy(redirect_logger, app.conf.worker_redirect_stdouts_level)
-            with contextlib.redirect_stdout(proxy), contextlib.redirect_stderr(proxy):
-
-                def alarm_handler(*args, **kwargs):
-                    raise SoftTimeLimitExceeded()
-
-                # Set alarm_handler to be called on a SIGALRM, and schedule a SIGALRM based on the soft time limit
-                signal.signal(signal.SIGALRM, alarm_handler)
-                signal.alarm(int(job_model.soft_time_limit) or settings.CELERY_TASK_SOFT_TIME_LIMIT)
-
-                try:
-                    eager_result = run_job.apply(
+        job_queue = JobQueue.objects.get(name=task_queue)
+        # Kubernetes Job Queue logic
+        if job_queue.queue_type == JobQueueTypeChoices.TYPE_KUBERNETES:
+            config_file_path = "./development/kind-kube-config"
+            config.load_kube_config(config_file_path)
+            core_v1 = core_v1_api.CoreV1Api()
+            api_instance = core_v1
+            name = "nautobot-pod-demo"
+            resp = None
+            try:
+                resp = api_instance.read_namespaced_pod(name=name, namespace="default")
+                print("Found old pod")
+            except ApiException as e:
+                if e.status != 404:
+                    print(f"Unknown error: {e}")
+                    sys.exit(1)
+
+            if not resp:
+                print(f"Pod {name} does not exist. Creating it...")
+                pod_manifest = {
```

Comment:
Should we instead assume that the pod manifest is part of the environment setup, rather than trying to guess the correct manifest here?

Reply:
I am playing around with a pod.yaml file that should replace the hardcoded `pod_manifest` here in this [story](https://app.zenhub.com/workspaces/nautobot-core-engineering-64a875e74842500d9e80e23e/issues/gh/nautobot/nautobot/6088)